### DISCUSSION TITLE
ওপেন বাংলা কি-বোর্ডে preedic অপশন বন্ধ করতে চাই।

### DISCUSSION THREAD

**Discussion by Mahmudul-hasan613** [OpenBangla/OpenBangla-Keyboard#296] [General]
ওপেন বাংলা কি-বোর্ড দিয়ে বাংলা টাইপিং করার সময় নিচে একটা আন্ডারলাইনের মতো কিছু একটা দেখা যায়। যেটা টাইপিং করার সময় সমস্যা সৃষ্টি করে। আমি এই ফিচারটি ibus থেকে বন্ধ করে দিতে চাই। টার্মিনালে ibus read-config লিখলে একটা কনফিগারেশন ফাইল দেখা যায়। সেখানে embed-preedic-text: true দেওয়া আছে। যেটা আমি false করে দিতে চাচ্ছি। কিন্তু কিভাবে সেটা false করতে হয় তা আমি খুঁজে পাচ্ছি না। যদি এব্যাপারে আমাকে সহযোগিতা করা হয়; তাহলে খুব উপকার হতো।
![Screenshot from 2022-06-26 23-14-56](https://user-images.githubusercontent.com/60663471/175826872-9ce17f76-8f12-45d0-b584-725733368afd.png)

**Comment by ahmubashshir** [comment]
> ওপেন বাংলা কি-বোর্ড দিয়ে বাংলা টাইপিং করার সময় নিচে একটা আন্ডারলাইনের মতো কিছু একটা দেখা যায়। যেটা টাইপিং করার সময় সমস্যা সৃষ্টি করে। আমি এই ফিচারটি ibus থেকে বন্ধ করে দিতে চাই। টার্মিনালে ibus read-config লিখলে একটা কনফিগারেশন ফাইল দেখা যায়। সেখানে embed-preedic-text: true দেওয়া আছে। যেটা আমি false করে দিতে চাচ্ছি। কিন্তু কিভাবে সেটা false করতে হয় তা আমি খুঁজে পাচ্ছি না। যদি এব্যাপারে আমাকে সহযোগিতা করা হয়; তাহলে খুব উপকার হতো। ![Screenshot from 2022-06-26 23-14-56](https://user-images.githubusercontent.com/60663471/175826872-9ce17f76-8f12-45d0-b584-725733368afd.png)


You can use `gsettings`/`dconf`  to change ibus settings.
eg:
```bash
$ gsettings get org.freedesktop.ibus.general embed-preedit-text
true
$ gsettings set org.freedesktop.ibus.general embed-preedit-text false
$ gsettings get org.freedesktop.ibus.general embed-preedit-text
false
```

**Reply by Mahmudul-hasan613** [reply]
ওপেন বাংলা কি-বোর্ড দিয়ে টাইপিং করার সময়, অক্ষরের নিচে যে একটা আন্ডার লাইনের মতো কিছু একটা দেখায়; সেটা বন্ধ করার কি কোনো অপশন নাই....?
![Screenshot from 2022-06-27 19-40-20](https://user-images.githubusercontent.com/60663471/175955663-55b9e285-3976-445b-9003-d13aca909d94.png)

**Reply by mominul** [reply]
দুঃখিত, আন্ডারলাইন অফ করার অপশন নেই এখনো পর্যন্ত, তবে এরকম অপশন যোগ করা যায় কিনা বিবেচনা করা হবে।

**Reply by Mahmudul-hasan613** [reply]
Show suggestions off করা থাকলে, ওপেন বাংলা কি-বোর্ড দিয়ে লেখার সময় আন্ডারলাইন দেখায় না। তবে লিব্রি অফিস রাইটারে ওয়ার্ড সাজেশন বন্ধ থাকলেও টাইপিং করার সময় আন্ডারলাইন দেখায়। যেটা লিব্রি অফিসের সমস্যা।

এছাড়া আমার আরেকটি বিষয় জানার ছিলো :
ওপেন বাংলা কি-বোর্ডের ৩.০.০ সংস্করণ কবে মুক্তি পাবে, এর সম্ভাব্য তারিখ কি ঠিক হয়েছে? কারণ আমি ওপেন বাংলা কি-বোর্ডের নতুন সংস্করণ ব্যবহার করার জন্য অধীর আগ্রহে অপেক্ষা করছি।

**Reply by mominul** [reply]
> ওপেন বাংলা কি-বোর্ডের ৩.০.০ সংস্করণ কবে মুক্তি পাবে, এর সম্ভাব্য তারিখ কি ঠিক হয়েছে? কারণ আমি ওপেন বাংলা কি-বোর্ডের নতুন সংস্করণ ব্যবহার করার জন্য অধীর আগ্রহে অপেক্ষা করছি।

দুঃখিত, কোন ডেডলাইন দিতে পারছি না, কিন্তু চেষ্টা করব জুলাই আগস্টের মধ্যে রিলিজ দিতে পারি কিনা। 
৩.০.০ সংস্করণের কোড [`develop`](https://github.com/OpenBangla/OpenBangla-Keyboard/tree/develop) ব্রাঞ্চে আছে, আপনি চাইলে কম্পাইল করে ব্যবহার করতে পারবেন।

**Reply by Mahmudul-hasan613** [reply]
উবুন্তুর ২২.০৪ সংস্করণের জন্য কম্পাইল করার পদ্ধতিটা বলে দিলে, আমার জন্য ইন্সটল করা সুবিধা হতো।

**Reply by Mahmudul-hasan613** [reply]
>  ৩.০.০ সংস্করণের কোড [`develop`](https://github.com/OpenBangla/OpenBangla-Keyboard/tree/develop) ব্রাঞ্চে আছে, আপনি চাইলে কম্পাইল করে ব্যবহার করতে পারবেন।

কম্পাইল করার সময় একটি সমস্যার সম্মুখিন হচ্ছি! cmake .. কমান্ডটি কাজ করছে না।
![Screenshot from 2022-07-12 22-01-25](https://user-images.githubusercontent.com/60663471/178539880-0462c12e-13e1-4336-ad69-1d0e1aa0fba0.png)

CMakeOutput.log
The system is: Linux - 5.15.0-41-generic - x86_64

**Reply by mominul** [reply]
প্রথমে ওপেনবাংলা কিবোর্ড আন-ইন্সটল করুন যদি ইন্সটল করে থাকেন।
এখন এই কমান্ডগুলো দিতে থাকুন terminal এঃ
```
sudo apt-get install build-essential rustc cargo cmake libibus-1.0-dev qt5-default libzstd-dev
git clone --recursive https://github.com/OpenBangla/OpenBangla-Keyboard.git
cd OpenBangla-Keyboard
git checkout develop
git submodule update
mkdir build && cd build
cmake .. -DENABLE_IBUS=ON
make
sudo make install
```

**Reply by Mahmudul-hasan613** [reply]
> sudo apt-get install build-essential rustc cargo cmake libibus-1.0-dev qt5-default libzstd-dev

qt5-default ইন্সটল করতে পারছি না!
![Screenshot from 2022-07-12 23-00-37](https://user-images.githubusercontent.com/60663471/178550746-20d2a5c7-2548-46d6-b026-cb64860ed74d.png)

**Reply by Mahmudul-hasan613** [reply]
> cmake .. -DENABLE_IBUS=ON

উপরের কমান্ড দিলে Error আসে!
![Screenshot from 2022-07-12 23-04-56](https://user-images.githubusercontent.com/60663471/178551497-d19ac002-8ec7-479b-a127-5adc1389707e.png)
![Screenshot from 2022-07-12 23-05-41](https://user-images.githubusercontent.com/60663471/178551571-70ecf53c-ea30-4a10-a4df-78f624acecf4.png)

**Reply by mominul** [reply]
পূর্বের `OpenBangla-Keyboard` নামের ফোল্ডারটি ডিলিট করুন এবং নতুন করে চেষ্টা করুন কমান্ড গুলো
```
sudo apt-get install build-essential rustc cargo cmake libibus-1.0-dev qtbase5-dev qtbase5-dev-tools libzstd-dev
git clone --recursive https://github.com/OpenBangla/OpenBangla-Keyboard.git
cd OpenBangla-Keyboard
git checkout develop
git submodule update
mkdir build && cd build
cmake .. -DENABLE_IBUS=ON
make
sudo make install
```

**Reply by Mahmudul-hasan613** [reply]
আমি ৩.০.০ develop সংস্করণটি ইন্সটল করেছি। সেখানে এখনো  [Preedit disappears and appears elsewhere when the cursor is moved with a mouse #298](https://github.com/OpenBangla/OpenBangla-Keyboard/issues/298) বাগটি রয়ে গেছে।

[Screencast from 13-7-22 09:32:47 পূর্বাহ্ণ +06.webm](https://user-images.githubusercontent.com/60663471/178647236-c0c58712-7308-4064-802e-11817bd32312.webm)

**Comment by ahmubashshir** [comment]
On Mon, Jun 27 2022 at 10:11:32 AM -0700, Mahmudul-hasan613 
***@***.***> wrote:
> ওপেন বাংলা কি-বোর্ড দিয়ে 
> টাইপিং করার সময়, অক্ষরের 
> নিচে যে একটা আন্ডার 
> লাইনের মতো কিছু একটা 
> দেখায়; সেটা বন্ধ করার কি 
> কোনো অপশন নাই....?

This is IBus bug. report it there.

	Mubashshir

**Comment by Mahmudul-hasan613** [comment]
ওপেন বাংলা কি-বোর্ডের জন্য ibus থেকে fcitx এ সুইচ করবো কিভাবে...?
